Add firmware settings validation reference (#111)
meshtastic/meshtastic-design: standards/audits/data/settings-validation-firmware.md

| `pa_fan_disabled` | `false` | **none** | |
| `ignore_incoming` | empty | **none** | nanopb limit 3 entries. |
| `serial_hal_only` | `false` | **none** | |

**Region/preset validation** is performed by `RadioInterface::validateConfigRegion()`
and `checkOrClampConfigLora()`. A key firmware behaviour: an invalid LoRa config
from the **local client** is **clamped** to working values, but the same config
from a **remote admin node** is **rejected** (reverted) — a remote node must not
be able to knock a device onto incompatible radio parameters.

### Bluetooth (`Config.BluetoothConfig`)

Every field is copied verbatim — no firmware-side validation.

| Field | Default | Validation | Notes |
|-------|---------|------------|-------|
| `enabled` | `true` | **none** | `false` on TFT/MUI boards. |
| `mode` | `RANDOM_PIN` | **none** | `FIXED_PIN` on screenless boards. |
| `fixed_pin` | `123456` | **none** | **Not** range-checked or digit-counted by the firmware — any `uint32` is accepted. |

### Security (`Config.SecurityConfig`)

| Field | Default | Validation | Notes |
|-------|---------|------------|-------|

> | `public_key` | empty | **sanitize** | Regenerated from the private key when a valid region is set. |

Author: Why sanitize public key? It's PUBLIC

| `enabled`, `allow_input_source` | `false` / `""` | **none** | Deprecated. |

The canned-message **text** is a separate `CannedMessageModuleConfig.messages`
string, default `"Hi|Bye|Yes|No|Ok"`, set via `set_canned_message_module_messages`
— `strncpy` into the buffer, **no** content check or UTF-8 sanitization.

### Audio (`ModuleConfig.AudioConfig`)

All defaults `0` / `false`. No firmware-side validation.

| Field | Default | Validation |
|-------|---------|------------|
| `codec2_enabled` | `false` | **none** |
| `ptt_pin`, `i2s_ws`, `i2s_sd`, `i2s_din`, `i2s_sck` | `0` | **none** |
| `bitrate` | `CODEC2_DEFAULT` | **none** |

### Remote Hardware (`ModuleConfig.RemoteHardwareConfig`)

All defaults `0` / `false`. No firmware-side validation.

| Field | Default | Validation | Notes |
|-------|---------|------------|-------|
| `enabled`, `allow_undefined_pin_access` | `false` | **none** | |
| `available_pins` | empty | **none** | nanopb limit 4 entries. |

### Neighbor Info (`ModuleConfig.NeighborInfoConfig`)

| Field | Default | Validation | Notes |
|-------|---------|------------|-------|
| `enabled` | `false` | **none** | |

> | `update_interval` | `0` | **clamp** | If below `14400` (`min_neighbor_info_broadcast_secs`) it is reset to `21600` (`default_neighbor_info_broadcast_secs`). |

Reviewer: ## Bug: This clamp is dead code in the firmware

The `AdminModule.cpp` handler (lines 1013–1020) checks and clamps the **old stored value**, then immediately overwrites the entire struct with the inbound payload:

```cpp
case meshtastic_ModuleConfig_neighbor_info_tag:
    moduleConfig.has_neighbor_info = true;
    if (moduleConfig.neighbor_info.update_interval < min_neighbor_info_broadcast_secs) {
        moduleConfig.neighbor_info.update_interval = default_neighbor_info_broadcast_secs;
    }
    moduleConfig.neighbor_info = c.payload_variant.neighbor_info; // ← overwrites the clamped value
    break;
```

The guard never takes effect — any inbound `update_interval` (including values below 14400) is stored verbatim.

```suggestion
| `update_interval` | `0` | **none** (intended clamp is dead code) | Firmware checks the *previous* stored value then overwrites with the inbound struct — the intended floor of `14400` (`min_neighbor_info_broadcast_secs`) never takes effect. See `AdminModule.cpp` lines 1016–1020. |
```

Reviewer: @thebentern - this is the call-out i was most concerned over

### Status Message (`ModuleConfig.StatusMessageConfig`)

| Field | Default | Validation |
|-------|---------|------------|
| `node_status` | `""` | **none** |

### Traffic Management (`ModuleConfig.TrafficManagementConfig`)

All 14 fields default proto-zero. No firmware-side validation. `Default.h`
defines intended runtime defaults (`position_precision_bits` 24,
`position_min_interval_secs` 43200) that the module applies at runtime when the
stored field is `0` — they are **not** written into the stored config.

| Field | Default | Validation |
|-------|---------|------------|
| `enabled`, `position_dedup_enabled`, `nodeinfo_direct_response`, `rate_limit_enabled`, `drop_unknown_enabled`, `exhaust_hop_telemetry`, `exhaust_hop_position`, `router_preserve_hops` | `false` | **none** |
| `position_precision_bits`, `position_min_interval_secs`, `nodeinfo_direct_response_max_hops`, `rate_limit_window_secs`, `rate_limit_max_packets`, `unknown_packet_threshold` | `0` | **none** |

### TAK (`ModuleConfig.TAKConfig`)

| Field | Default | Validation | Notes |
|-------|---------|------------|-------|

> | `team` | `Unspecifed_Color` (firmware renders Cyan) | **none** (no-op) | `handleSetModuleConfig()` has **no `case` for `tak`** — a `set_module_config` carrying the `tak` variant is silently a no-op (nothing stored, no error returned). TAK config cannot be set via `set_module_config` in this firmware revision. |

Reviewer: Confirmed accurate — no `meshtastic_ModuleConfig_tak_tag` case exists in the switch. Worth noting that `get_module_config_request` for TAK *does* respond correctly (the response switch at line 1236 is missing it too actually — let me double-check... actually the response handler has no `TAK_CONFIG` case either in the `get_module_config_request` handler). So TAK config is fully inert in both directions via `AdminModule`. Good as documented.

**Minor nit:** Consider noting that the `ModuleConfigType` enum *does* define `TAK_CONFIG = 15` in the proto, so clients can send the request — it just does nothing firmware-side.

| `role` | `Unspecifed` (firmware renders TeamMember) | **none** (no-op) | Same. |

---

## Channel Config

Inbound via `set_channel` → `handleSetChannel()` → `Channels::setChannel()`.
A factory device exposes 8 channel slots (`MAX_NUM_CHANNELS`): index 0 is the
populated primary, indexes 1–7 are `DISABLED` with empty settings.

| Field | Default (channel 0) | Validation | Notes |
|-------|----------------------|------------|-------|
| `index` | `0` | **reject** | Must be `0 ≤ index < 8`; otherwise `BAD_REQUEST`. `get_channel_request` is rejected the same way. |
| `role` | `PRIMARY` | **sanitize** | Setting a channel to `PRIMARY` demotes every other `PRIMARY` channel to `SECONDARY`. |
| `name` | `""` | **none** | nanopb limit 12 bytes. Empty ⇒ rendered as the modem-preset name. |

> | `psk` | `{0x01}` (default-key shorthand) | **none** | Not validated at write time. `getKey()` later interprets it: `0` = off, `1`–`10` = built-in keys, 16 / 32 bytes = AES-128 / AES-256. A short key (2–15 or 17–31 bytes) is silently **zero-padded** to 16 / 32 — not rejected. |

Reviewer: Accurate but could note the timing distinction for SDK implementors:

```suggestion
| `psk` | `{0x01}` (default-key shorthand) | **none** | Not validated at write time; stored verbatim. `getKey()` at *read time* interprets it: `0` = off, `1`–`10` = built-in keys, 16 / 32 bytes = AES-128 / AES-256. A short key (2–15 or 17–31 bytes) is silently **zero-padded** to 16 / 32 at use time — not rejected or transformed on write. |
```

| `telemetry.environment_measurement_enabled` | — | — | `true` | — | — | — | — | — |
| `owner.is_unmessagable` | `true` | `true` | `true` | `true` | — | `true` | — | — |

`ROUTER` also re-runs the interval initializers, flipping every `IF_ROUTER`
default (`gps_update_interval`, `ls_secs`, `sds_secs`, `wait_bluetooth_secs`,
`screen_on_secs`) to its router branch.

---

## Authorization & Session Passkey

Before any `set_*` is applied, `handleReceivedProtobuf()` runs an authorization
gate — the firmware analog of "who is allowed to write settings at all":

1. **Local origin** (connected client) — allowed unless `security.is_managed`.
2. **Legacy admin channel** — allowed only if `security.admin_channel_enabled`.
3. **PKI-authenticated remote** — the sender's public key must match one of the
   up-to-three 32-byte `security.admin_key[]` entries; otherwise
   `ADMIN_PUBLIC_KEY_UNAUTHORIZED`.
4. Anything else ⇒ `NOT_AUTHORIZED`.

**Session passkey** — a state-changing message from a *remote* node must echo
back an 8-byte `session_passkey` issued on a prior `get_*` response. It is valid

> for 300 s; a mismatch or timeout ⇒ `ADMIN_BAD_SESSION_KEY`. Locally-connected

Reviewer: Two precision improvements here:

1. The passkey check condition is `mp.from != 0 && !messageIsRequest(r) && !messageIsResponse(r)` — so remote `get_*` requests (classified as "requests" by `messageIsRequest()`) also skip the check, not just locally-connected clients.

2. The passkey is included in **every** admin `get_*` response (owner, config, module_config, channel, metadata, connection_status, remote_hardware_pins, plus module API responses), not just a single prior response. It regenerates only if 150+ seconds have elapsed since last generation, creating an overlapping validity window (regenerates at 150s, valid until 300s).

```suggestion
for 300 s; a mismatch or timeout ⇒ `ADMIN_BAD_SESSION_KEY`. Locally-connected
clients skip this check, as do remote `get_*` requests (only non-request,
non-response admin messages are gated). The passkey is regenerated after 150 s
but remains valid for 300 s, creating an overlapping validity window.
```
